### DISCUSSION TITLE
"Containing" Bodies?

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#165] [Q&A]
I have a few scenarios where I would like to keep bodies inside a bound area.
To do this, I create 6 shapes around my inside box, or in the case of a cylinder I create my own inside shape and then add a roof/floor.
![surroundingShapes](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/72bee751-bb69-40cd-b24d-70b327d11090)

But I'm wondering if there was a way I could just use standard shapes and REVERSE the collision system to keep things INSIDE the body.
I've done this with making my own primitive physics systems but that was 2D and a long time ago. 

I'm imagining a shape bouncing off the insides of a box or cylinder
![boundShapes](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/163c1054-37f2-4337-8d13-c48ff5583433)

I'm not sure this is even possible but I figured someone might have some ideas and I would appreciate any insights.

**Comment by jrouwe** [comment]
Jolt currently has no way of inverting a shape (and I also don't see an easy way of implementing this as it changes convex shapes into non-convex shapes). I would suggest creating a compound of convex shapes to approximate your 'interior' or to use a mesh shape.
